sdl2promela/stop_condition/translator.py

```diff
@@ -0,0 +1,359 @@
+        .build()
+    )
+
+
+@dispatch(model.PlusExpression)
+def _generate(expr: model.PlusExpression):
+    return (
+        promelaBuilder.BinaryExpressionBuilder(promela.BinaryOperator.ADD)
+        .withLeft(_generate(expr.lhs))
+        .withRight(_generate(expr.rhs))
+        .build()
+    )
+
+
+@dispatch(model.MinusExpression)
+def _generate(expr: model.MinusExpression):
+    return (
+        promelaBuilder.BinaryExpressionBuilder(promela.BinaryOperator.SUBTRACT)
+        .withLeft(_generate(expr.lhs))
+        .withRight(_generate(expr.rhs))
+        .build()
+    )
+
+
+@dispatch(model.MulExpression)
+def _generate(expr: model.MulExpression):
+    return (
+        promelaBuilder.BinaryExpressionBuilder(promela.BinaryOperator.MULTIPLY)
+        .withLeft(_generate(expr.lhs))
+        .withRight(_generate(expr.rhs))
+        .build()
+    )
+
+
+@dispatch(model.DivExpression)
+def _generate(expr: model.DivExpression):
+    return (
+        promelaBuilder.BinaryExpressionBuilder(promela.BinaryOperator.DIVIDE)
+        .withLeft(_generate(expr.lhs))
+        .withRight(_generate(expr.rhs))
+        .build()
+    )
+
+
+@dispatch(model.ModExpression)
+def _generate(expr: model.ModExpression):
+    return (
+        promelaBuilder.BinaryExpressionBuilder(promela.BinaryOperator.MODULO)
+        .withLeft(_generate(expr.lhs))
+        .withRight(_generate(expr.rhs))
+        .build()
+    )
+
+
+@dispatch(model.NegationExpression)
+def _generate(expr: model.NegationExpression):
+    return (
+        promelaBuilder.UnaryExpressionBuilder(promela.UnaryOperator.NEGATIVE)
+        .withExpression(expr.expression)
+        .build()
+    )
+
+
+@dispatch(model.IntegerValue)
+def _generate(expr: model.IntegerValue):
+    return promela.IntegerValue(expr.value)
+
+
+@dispatch(model.FloatValue)
+def _generate(expr: model.FloatValue):
+    return promela.FloatValue(expr.value)
+
+
+@dispatch(model.BooleanValue)
+def _generate(expr: model.BooleanValue):
+    return promela.BooleanValue(expr.value)
+
+
+@dispatch(model.VariableReference)
+def _generate(expr: model.VariableReference):
+    return promelaBuilder.VariableReferenceBuilder(expr.name).build()
+
+
+@dispatch(model.Selector)
+def _generate(expr: model.Selector):
+
+    promelaObjects = [_generate(elem) for elem in expr.elements]
+
+    if not isinstance(promelaObjects[0], promela.VariableReference):
+        raise TranslateException(
+            "Call Expression is not allowed as first element of selector."
+        )
+
+    result: promela.MemberAccess = (
+        promelaBuilder.MemberAccessBuilder()
+        .withUtypeReference(
+            promelaBuilder.MemberAccessBuilder()
+            .withUtypeReference(
+                promelaBuilder.VariableReferenceBuilder("global_state").build()
+            )
+            .withMember(promelaObjects[0])
+            .build()
+        )
+        .withMember(promelaObjects[1])
+        .build()
+    )
+    promelaObjects.pop(0)
```

Comment:
I assume that the first element must be accessed via global_state, hence the result initialiation and 1 pop. Why 2 pops? Couldn't this be handled in the later loop? Also, a comment clarifying the intents would be nice.